Code/Framework/AzFramework/AzFramework/Spawnable/Spawnable.h

```diff
@@ -29,7 +30,143 @@ namespace AzFramework
+        struct EntityAlias
+        {
+            AZ_CLASS_ALLOCATOR(EntityAlias, AZ::SystemAllocator, 0);
+            AZ_TYPE_INFO(AzFramework::Spawnable::EntityAlias, "{C8D0C5BC-1F0B-4572-98C1-73B2CA8C9356}");
+
+            bool HasLowerIndex(const EntityAlias& other) const;
+
+            AZ::Data::Asset<Spawnable> m_spawnable; //!< The spawnable containing the target entity to spawn.
+            uint32_t m_tag{ 0 }; //!< A unique tag to identify this alias with.
+            uint32_t m_sourceIndex{ 0 }; //!< The index of the entity in the original spawnable that will be replaced.
+            uint32_t m_targetIndex{ 0 }; //!< The index of the entity in the target spawnable that will be used to replace the original.
+            EntityAliasType m_aliasType{ EntityAliasType::Original }; //!< The kind of replacement.
+            bool m_queueLoad{ false }; //!< Whether or not to automatically queue the spawnable for loading.
+        };
+
         using EntityList = AZStd::vector<AZStd::unique_ptr<AZ::Entity>>;
+        using EntityAliasList = AZStd::vector<EntityAlias>;
+
+    private:
+        class EntityAliasVisitorBase
+        {
+        protected:
+            bool IsSet(const EntityAliasList* aliases) const;
+
+            bool HasAliases(const EntityAliasList* aliases) const;
+            bool AreAllSpawnablesReady(const EntityAliasList* aliases) const;
+
+            EntityAliasList::const_iterator begin(const EntityAliasList* aliases) const;
+            EntityAliasList::const_iterator end(const EntityAliasList* aliases) const;
+            EntityAliasList::const_iterator cbegin(const EntityAliasList* aliases) const;
+            EntityAliasList::const_iterator cend(const EntityAliasList* aliases) const;
+
+            using ListTargetSpawanblesCallback = AZStd::function<void(const AZ::Data::Asset<Spawnable>& targetSpawnable)>;
+            void ListTargetSpawnables(const EntityAliasList* aliases, const ListTargetSpawanblesCallback& callback) const;
+            void ListTargetSpawnables(const EntityAliasList* aliases, AZ::Crc32 tag, const ListTargetSpawanblesCallback& callback) const;
+        };
+
+    public:
+        class EntityAliasVisitor final : public EntityAliasVisitorBase
+        {
+        public:
+            EntityAliasVisitor(Spawnable& owner, EntityAliasList* m_entityAliasList);
+            ~EntityAliasVisitor();
+
+            EntityAliasVisitor(EntityAliasVisitor&& rhs);
+            EntityAliasVisitor& operator=(EntityAliasVisitor&& rhs);
+
+            EntityAliasVisitor(const EntityAliasVisitor& rhs) = delete;
+            EntityAliasVisitor& operator=(const EntityAliasVisitor& rhs) = delete;
+
+            //! Checks if the visitor was able to retrieve data. This needs to be checked before calling any other functions.
+            bool IsSet() const;
+
+            bool HasAliases() const;
+            bool AreAllSpawnablesReady() const;
+
+            EntityAliasList::const_iterator begin() const;
```

Comment:
A guide comment for "the list itself is immutable during iteration so these are const iterators" would prevent some light astonishment here